crates/transaction-pool/src/blobstore/helpers.rs

```diff
@@ -0,0 +1,21 @@
+use alloy_eips::eip4844::{BlobAndProofV1, BlobTransactionSidecar};
+use alloy_primitives::B256;
+
+/// Helper function to match versioned hashes and construct `BlobAndProofV1` objects.
+pub(crate) fn match_versioned_hashes(
+    blob_sidecar: &BlobTransactionSidecar,
+    versioned_hashes: &[B256],
+) -> Vec<Option<BlobAndProofV1>> {
```

Comment:
could you perhaps also upstream this to alloy and make this a function of `BlobTransactionSidecar`?